packages/flutter_tools/doc/attach.md

```diff
@@ -0,0 +1,33 @@
+# Flutter Attach
+
+## Overview
+
+A Flutter-command that attaches to applications that have been launched
+without `flutter run` and provides a HotRunner (enabling hot reload/restart).
+
+## Usage
+
+There are four ways for the attach command to discover a running app:
+
+1. If the app is already running and the observatory port is known, it can be
+explicitly provided to attach via the command-line, e.g. `$ flutter attach
+--debug-port 12345`
+1. If the app is already running and the platform is iOS, attach can use mDNS
+to lookup the observatory port via the application ID, with just `$ flutter
+attach`
+1. If the platform is Fuchsia and the app is not yet running, the module name
+must be provided, e.g. `$ flutter attach --module=mod_name`. Attach will then
+listen and wait for the mod to start
+1. On other platforms (i.e. Android), if the app is not yet running attach
+will listen and wait for the app to be (manually) started with the default
+command: `$ flutter attach`
+
+## To-do
```

Comment:
If we don't already have an issue filed for this, then that's where this should go instead of here.

Reply:
We have [this issue](https://github.com/flutter/flutter/issues/23687), so I will remove this text.